src/day15.scala

```diff
@@ -0,0 +1,69 @@
+// using scala 3.1.0
+
+package day15
+
+import scala.util.Using
+import scala.io.Source
+import scala.collection.mutable
+
+type Coord = (Int, Int)
+class GameMap(cells: Seq[Seq[Int]]):
```

Comment:
```suggestion
class GameMap(cells: IndexedSeq[IndexedSeq[Int]]):
```
It's very important for this algorithm that indexed access be O(1) on `cells`, so we should require that in the type.

Reply:
Okay, it seems `List` is not an `IndexedSeq`. Once I made it, it seems we won an additional second or so of runtime. Nice!